sklearn/linear_model/ridge.py

```diff
@@ -367,11 +367,28 @@ def _ridge_regression(X, y, alpha, sample_weight=None, solver='auto',
                       return_n_iter=False, return_intercept=False,
                       X_scale=None, X_offset=None):
 
-    if return_intercept and sparse.issparse(X) and solver != 'sag':
-        if solver != 'auto':
-            warnings.warn("In Ridge, only 'sag' solver can currently fit the "
-                          "intercept when X is sparse. Solver has been "
-                          "automatically changed into 'sag'.")
+    has_sw = sample_weight is not None
+
+    if solver == 'auto':
+        if return_intercept:
+            # only sag supports fitting intercept directly
+            solver = "sag"
+        elif has_sw:
```

Comment:
I tought 'saga' also supports fitting intercept

Reply:
I tried fitting intercept with saga, and I get a warning in test `test_ridge_fit_intercept_sparse` with `solver='saga'`:

```ConvergenceWarning('The max_iter was reached which means the coef_ did not converge')```

so I guess it's not supported.